client/disputes/evidence/index.js

```diff
@@ -353,252 +47,85 @@ const getDisputeProductType = ( dispute ) => {
 
 // Temporary MVP data wrapper
 export default ( { query } ) => {
-	const path = `/wc/v3/payments/disputes/${ query.id }`;
+	const { id: disputeId } = query;
+	const { dispute, isLoading, updateDispute } = useDispute( disputeId );
 
-	const [ dispute, setDispute ] = useState();
-	const [ loading, setLoading ] = useState( false );
-	const [ evidence, setEvidence ] = useState( {} ); // Evidence to update.
 	const {
-		createSuccessNotice,
-		createErrorNotice,
-		createInfoNotice,
-	} = useDispatch( 'core/notices' );
-
-	const pristine =
-		! dispute ||
-		isMatchWith(
-			dispute.evidence,
-			evidence,
-			( disputeValue, formValue ) => {
-				// Treat null and '' as equal values.
-				if ( null === disputeValue && ! formValue ) {
-					return true;
-				}
-			}
-		);
-
-	const confirmationNavigationCallback = useConfirmNavigation( () => {
-		if ( pristine ) {
-			return;
-		}
-
-		return __(
-			'There are unsaved changes on this page. Are you sure you want to leave and discard the unsaved changes?',
-			'woocommerce-payments'
-		);
-	} );
-
-	useEffect( confirmationNavigationCallback, [
-		pristine,
-		confirmationNavigationCallback,
-	] );
+		isSavingEvidence,
+		isUploadingEvidence,
+		evidenceTransient,
+		evidenceUploadErrors,
+		saveEvidence,
+		submitEvidence,
+		updateEvidenceTransientForDispute,
+		updateEvidenceUploadErrorsForDispute,
+		uploadFileEvidenceForDispute,
+	} = useDisputeEvidence( disputeId );
+
+	const { setMessage: setNavigationMessage } = useConfirmNavigation();
 
 	useEffect( () => {
-		const fetchDispute = async () => {
-			setLoading( true );
-			try {
-				setDispute( await apiFetch( { path } ) );
-			} finally {
-				setLoading( false );
-			}
-		};
-
-		fetchDispute();
-	}, [ setLoading, setDispute, path ] );
-
-	const updateEvidence = ( key, value ) =>
-		setEvidence( ( e ) => ( { ...e, [ key ]: value } ) );
-	const updateDispute = ( updates = {} ) =>
-		setDispute( ( d ) => merge( {}, d, updates ) );
-	const isUploadingEvidence = () => some( dispute.isUploading );
-
-	const doRemoveFile = ( key ) => {
-		updateEvidence( key, '' );
-		updateDispute( {
-			metadata: { [ key ]: '' },
-			uploadingErrors: { [ key ]: '' },
-			fileSize: { [ key ]: 0 },
-		} );
-	};
+		const isPristine =
+			! dispute ||
+			! evidenceTransient || // Empty evidence transient means no local updates.
+			isMatchWith(
+				dispute.evidence,
+				evidenceTransient,
+				( disputeValue, formValue ) => {
+					// Treat null and '' as equal values.
+					if ( null === disputeValue && ! formValue ) {
+						return true;
+					}
+				}
+			);
```

Comment:
I moved the `isPristine` check into a `useEffect` hook to make it a bit clearer what sort of changes are supposed to update this value (note the dependencies array), but that change isn't strictly necessary.